Sources/MapboxCoreNavigation/RerouteController.swift

```diff
@@ -0,0 +1,211 @@
+                customRoutingProvider.map { _ in self } ?? defaultRerouteController
+            )
+        }
+    }
+    
+    private var config: NavigatorConfig
+    
+    // MARK: Reporting Data
+    
+    weak var delegate: ReroutingObserverDelegate?
+    
+    func userIsOnRoute() -> Bool {
+        return !rerouteDetector.isReroute()
+    }
+    
+    func forceReroute() {
+        rerouteDetector.forceReroute()
+    }
+    
+    // MARK: Internal State Management
+    
+    private let defaultRerouteController: RerouteControllerInterface
+    private let rerouteDetector: RerouteDetectorInterface
+    
+    private var reroutingRequest: NavigationProviderRequest?
+    private var recentRouteResponse: (response: RouteResponse, options: RouteOptions)?
+    private var isCancelled = false
+    
+    private weak var navigator: MapboxNavigationNative.Navigator?
+    
+    func resetToDefaultSettings() {
+        reroutesProactively = true
+        isCancelled = false
+        config.avoidManeuverSeconds = NSNumber(value: Self.DefaultManeuverAvoidanceRadius)
+        customRoutingProvider = nil
+    }
+    
+    required init(_ navigator: MapboxNavigationNative.Navigator, config: NavigatorConfig) {
+        self.navigator = navigator
+        self.config = config
+        self.defaultRerouteController = navigator.getRerouteController()
+        self.rerouteDetector = navigator.getRerouteDetector()
+        self.navigator?.addRerouteObserver(for: self)
+    }
+    
+    deinit {
+        self.navigator?.removeRerouteObserver(for: self)
+        self.navigator?.setRerouteControllerForController(defaultRerouteController)
+    }
+}
+
+extension RerouteController: RerouteObserver {
+    
+    private func decode(routeRequest: String) -> (routeOptions: RouteOptions, credentials: Credentials)? {
+        guard let requestURL = URL(string: routeRequest),
+              let routeOptions = RouteOptions(url: requestURL) else {
+                  return nil
+        }
+        return (routeOptions, Credentials(requestURL: requestURL))
+    }
+    
+    private func decode(routeResponse: String,
+                        routeOptions: RouteOptions,
+                        credentials: Credentials) -> RouteResponse? {
+        guard let data = routeResponse.data(using: .utf8) else {
+            return nil
+        }
+        
+        let decoder = JSONDecoder()
+        decoder.userInfo[.options] = routeOptions
+        decoder.userInfo[.credentials] = credentials
+        
+        return try? decoder.decode(RouteResponse.self,
+                                   from: data)
+    }
+    
+    func onSwitchToAlternative(forAlternativeRouteId alternativeRouteId: String,
+                               routeResponse: String,
+                               routeRequest: String,
+                               origin: RouterOrigin) {
+        // to be filled with Continuous Alternatives implementation
```

Comment:
Remember to ticket out this work. Is this no-op appropriate in the meantime?

Reply:
There is [a ticket](https://github.com/mapbox/navigation-sdks/issues/1376) already. Leaving this method empty just disables any alternative routes processing.